From 039ad756299384771c3c6d8f64cfaf8733e31c5c Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 5 Sep 2014 18:55:52 -0400
Subject: [scripts] build bin2c based on config option CONFIG_BUILD_BIN2C

Message-id: <1409943373-12669-18-git-send-email-vgoyal@redhat.com>
Patchwork-id: 91538
O-Subject: [RHEL7.1 PATCH 17/38] kernel: build bin2c based on config option CONFIG_BUILD_BIN2C
Bugzilla: 852066
RH-Acked-by: Dave Anderson <anderson@redhat.com>
RH-Acked-by: Baoquan He <bhe@redhat.com>
RH-Acked-by: Dave Young <dyoung@redhat.com>
RH-Acked-by: Chao WANG <chaowang@redhat.com>

BZ:852066

commit de5b56ba51f63973ceb5c184ee0855f0c8a13fc9
Author: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 8 Aug 2014 14:25:41 -0700

currently bin2c builds only if CONFIG_IKCONFIG=y. But bin2c will now be
used by kexec too.  So make it compilation dependent on CONFIG_BUILD_BIN2C
and this config option can be selected by CONFIG_KEXEC and CONFIG_IKCONFIG.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>
Cc: Borislav Petkov <bp@suse.de>
Cc: Michael Kerrisk <mtk.manpages@gmail.com>
Cc: Yinghai Lu <yinghai@kernel.org>
Cc: Eric Biederman <ebiederm@xmission.com>
Cc: H. Peter Anvin <hpa@zytor.com>
Cc: Matthew Garrett <mjg59@srcf.ucam.org>
Cc: Greg Kroah-Hartman <greg@kroah.com>
Cc: Dave Young <dyoung@redhat.com>
Cc: WANG Chao <chaowang@redhat.com>
Cc: Baoquan He <bhe@redhat.com>
Cc: Andy Lutomirski <luto@amacapital.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
index fdde1bc..174f855 100644
--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -1621,6 +1621,7 @@ source kernel/Kconfig.hz
 
 config KEXEC
  bool "kexec system call"
+ select BUILD_BIN2C
  ---help---
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/init/Kconfig b/init/Kconfig
index a541d35..cbaba6d 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -721,8 +721,13 @@ endchoice
 
 endmenu # "RCU Subsystem"
 
+config BUILD_BIN2C
+ bool
+ default n
+
 config IKCONFIG
  tristate "Kernel .config support"
+ select BUILD_BIN2C
  ---help---
    This option enables the complete Linux kernel ".config" file
    contents to be saved in the kernel. It provides documentation
diff --git a/scripts/basic/Makefile b/scripts/basic/Makefile
index afbc1cd..ec10d93 100644
--- a/scripts/basic/Makefile
+++ b/scripts/basic/Makefile
@@ -9,7 +9,7 @@
 # fixdep:   Used to generate dependency information during build process
 
 hostprogs-y := fixdep
-hostprogs-$(CONFIG_IKCONFIG)     += bin2c
+hostprogs-$(CONFIG_BUILD_BIN2C)     += bin2c
 always  := $(hostprogs-y)
 
 # fixdep is needed to compile other host programs
-- 
1.7.1